relayer-cli/src/commands/config/auto.rs

```diff
@@ -0,0 +1,93 @@
+use std::path::PathBuf;
+
+use abscissa_core::clap::Parser;
+use abscissa_core::{Command, Runnable};
+use chain_registry::{error::RegistryError, relayer_config::hermes_config};
+
+use crate::conclude::Output;
+
+use futures::{stream::FuturesUnordered, StreamExt};
+
+use ibc_relayer::config::{store, ChainConfig, Config};
+
+use tokio::runtime::Builder;
+
+#[derive(Clone, Command, Debug, Parser, PartialEq)]
+pub struct AutoCmd {
+    #[clap(
+        long = "path",
+        short = 'p',
+        required = true,
+        value_name = "PATH",
+        help_heading = "REQUIRED",
+        help = "Path to the configuration file"
+    )]
+    path: PathBuf,
+
+    #[clap(
+        long = "chains",
+        required = true,
+        multiple = true,
+        value_name = "CHAIN_1 CHAIN_2...",
+        help_heading = "REQUIRED",
+        help = "Identifier of the chains to include in the config"
+    )]
+    chain_ids: Vec<String>,
+
+    #[clap(
+        long = "keys",
+        required = true,
+        multiple = true,
+        value_name = "KEY_CHAIN_1 KEY_CHAIN_2...",
+        help_heading = "REQUIRED",
+        help = "Key names to include in the config"
+    )]
+    keys: Vec<String>,
+}
+
+async fn get_chain_configs(
+    chain_names: &[String],
+    keys: &[String],
+) -> Vec<Result<ChainConfig, RegistryError>> {
+    let futures: FuturesUnordered<_> = chain_names
+        .iter()
+        .zip(keys.iter())
+        .map(|(chain_name, key)| hermes_config(chain_name.as_str(), key.as_str()))
+        .collect();
+
+    futures
+        .collect::<Vec<Result<ChainConfig, RegistryError>>>()
+        .await
+}
+
+impl Runnable for AutoCmd {
+    fn run(&self) {
+        // Assert that for every chain, a key name is provided
+        if self.chain_ids.len() != self.keys.len() {
+            Output::error("Must provide a key name for every chain").exit();
+        }
+        let runtime = Builder::new_multi_thread()
+            .worker_threads(1)
```

Comment:
I think we can get away by using the single threaded Tokio runtime here:

```suggestion
        let runtime = Builder::new_current_thread()
```